### PR TITLE
Add CVO sync from payload

### PR DESCRIPTION
**Pull request by abhinavdahiya**
* adds funcs to load manifests from disk
* add resource builder that allows Do on any **lib.Manifest**
* adds syncUpdatePayloadContents that ensures <prefix>/<version>
  contains the <version> update payload
* adds syncUpdatePayload that uses lib.resourcebuilder to generically
  apply the update payload in order.

/assign @crawford @yifan-gu

### CODE REVIEW

File: lib/manifest.go [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+
+	obj *unstructured.Unstructured
+}
+
+// UnmarshalJSON unmarshals bytea of single kubernetes object to Manifest.
+func (m *Manifest) UnmarshalJSON(in []byte) error {
+	if m == nil {
+		return errors.New("Manifest: UnmarshalJSON on nil pointer")
+	}
+	if !bytes.Equal(in, []byte("null")) {
```
Comment (yifan-gu) [inline lib/manifest.go]: nit: why need to special case this? won't this be handled by the Decode()?

File: lib/manifest.go [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+	m.GVK = ud.GroupVersionKind()
+	m.obj = ud.DeepCopy()
+	return nil
+}
+
+// Object returns underlying metav1.Object
+func (m *Manifest) Object() metav1.Object { return m.obj }
+
+const (
+	rootDirKey = "000"
```
Comment (yifan-gu) [inline lib/manifest.go]: @abhinavdahiya Could you comment why we need this?

File: lib/manifest.go [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+//         manifest0
+//         manifest1
+//     01_subdir1/
+//         manifest0
+//         manifest1
+// LoadManifests(<abs path to>/root):
+// returns map
+// 000: [manifest0, manifest1]
+// 00_subdir0: [manifest0, manifest1]
+// 00_subdir0: [manifest0, manifest1]
```
Comment (yifan-gu) [inline lib/manifest.go]: this might be 01_subdir_1?

File: lib/manifest.go (hunk 1 above)
Comment (abhinavdahiya) [reply lib/manifest.go]: update it add comment. look [here](https://github.com/openshift/cluster-version-operator/pull/10/commits/109df8379ae64172ca9ecf4ce2061f71b2a8009c#diff-d1961d4f655d836d50969836a0f51785R96)

File: lib/manifest.go (hunk 2 above)
Comment (abhinavdahiya) [reply lib/manifest.go]: done

File: lib/manifest.go (hunk 3 above)
Comment (abhinavdahiya) [reply lib/manifest.go]: :+1:

File: lib/manifest.go [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+
+// Object returns underlying metav1.Object
+func (m *Manifest) Object() metav1.Object { return m.obj }
+
+const (
+	// rootDirKey is used as key for the manifest files in root dir
+	// passed to LoadManifests
+	// It is set to `000` to give it more priority if the actor sorts
+	// based on keys.
+	rootDirKey = "000"
```
Comment (smarterclayton) [inline lib/manifest.go]: Alternatively we could just force the root dir to be a real dir and ignore files in the root (treat as not manifests).  The only content we had in there in the payload right now was the image mapping and the Cincinnati file.

Comment (abhinavdahiya) [reply lib/manifest.go]: If the update payload is 
```console
/cincinnati.json
/images.json
manifests/
   ....
   ....
```
The root here would mean manifests dir

**Comment by abhinavdahiya** [comment]
@yifan-gu @smarterclayton any updates ??

File: lib/manifest.go (hunk 4 above)
Comment (yifan-gu) [reply lib/manifest.go]: +1 on keeping the manifests dir, seems pretty clean to me and not much overhead.

**Comment by yifan-gu** [comment]
/lgtm
Let go with it!

**Comment by smarterclayton** [comment]
I'm going to test the prototype release build tool against this image today.

**Pull request merged**